artifacts/otg.proto

```diff
@@ -7037,6 +7163,94 @@ message ActionProtocolBgpInitiateGracefulRestart {
   optional uint32 restart_delay = 2;
 }
 
+// Import V4 and V6 routes installed on and exported from a DUT,
+// the format for which could be different across vendors, and convert
+// those to OTG V4 or V6 routes configured across corresponding BGP peers.
+// Following additional details need to be considered:
+// - The exported routes may be provided in variety of file formats
+// - A given file may contain both V4 and V6 route entries
+// - Each route entry shall correspond to `Bgp.V4RouteRange` or `Bgp.V6RouteRange`
+// with exactly one entry of `V4RouteAddress` or `V6RouteAddress` respectively
+// - The name of resultant OTG V4 or V6 routes shall be a constituent of
+// an alias associated with import file, name of corresponding BGP peer
+// and the sequence number corresponding to its order of entry in the file
+message ActionProtocolBgpImportRoutes {
+
+  // Description missing in models
+  repeated ActionProtocolBgpRouteImport imports = 1;
+}
+
+// Configuration for specifying the format and corresponding characteristics of V4 or
+// V6 routes to be imported.
+message ActionProtocolBgpRouteImport {
+
+  // Globally unique name of an object. It also serves as the primary key for arrays of
+  // objects.
+  // required = true
+  string name = 1;
+
+  // Description missing in models
+  optional ActionProtocolBgpRoutes routes = 2;
+
+  // List of import targets (i.e. corresponding V4 or V6 route groups) and
+  // associated import properties
+  repeated ActionProtocolBgpRouteImportTarget targets = 3;
+}
+
+// Routes to be imported in a given format.
+message ActionProtocolBgpRoutes {
+
+  message Choice {
+    enum Enum {
+      unspecified = 0;
+      v4_routes = 1;
+      v6_routes = 2;
+      openconfig_routes = 3;
+      cisco_routes = 4;
+      juniper_routes = 5;
+    }
+  }
+  // Format of routes to be imported.
+  optional Choice.Enum choice = 1;
+
+  // Description missing in models
+  repeated BgpV4RouteRange v4_routes = 2;
+
+  // Description missing in models
+  repeated BgpV6RouteRange v6_routes = 3;
+
+  // Binary containing routes in OpenConfig format
```

Comment:
i am not clear why we are trying import these a byte blobs as part of OTG?

Shouldn't we have defined an OTG specific modelling for the route and then have a tool which can convert the vendor specific formats to the OTG native format

Reply:
We already have data model for representing routes to be imported in OTG:
- https://github.com/open-traffic-generator/models/blob/load-routes/artifacts/otg.proto#L2552
- https://github.com/open-traffic-generator/models/blob/load-routes/artifacts/otg.proto#L7201

And we indeed can just have a Go test where the community maintained Go library is used to convert vendor-specific exports to OTG routes as modeled above. But the challenges would be:
- We'll have to maintain the same library for different languages (e.g. python) for tests that can't use Go
- Users will manually have to do some transformations for some route parameters (e.g. next hop address)

Hence a more language agnostic way to achieve this would be to just feed the blob to ixia-c-controller and let it do the intended transcoding and transformation (while still using the community maintained Go library for transcoding).